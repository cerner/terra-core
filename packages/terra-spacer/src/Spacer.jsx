import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Spacer.module.scss';
import { mapShorthandToObject, shorthandValidator } from './_spacerShorthandUtils';
import { SpacerSizes, SpacerClassMappings } from './_spacerConstants';

const cx = classNames.bind(styles);

/*
  NOTE: this is needed in order to ensure that the types `marginTop`, `marginRight`, `paddingTop`, `paddingRight`, etc. are recognized by terra-props-table
  as defined. Currently, simply putting `Object.values(SpacerSizes)` causes all other `marginX` and `paddingX` props to be recognized as `undefined`
  in the PropsTable.
*/
const arrayOfSpacerSizes = ['none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'];

const propTypes = {
  /**
   * Child Nodes.
   */
  children: PropTypes.node.isRequired,
  /**
   * Sets margin in a syntax similar to the standard CSS spec https://developer.mozilla.org/en-US/docs/Web/CSS/margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  margin: shorthandValidator,
  /**
   * Sets top margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  marginTop: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets bottom margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  marginBottom: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets left margin One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  marginLeft: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets right margin. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  marginRight: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets padding in a syntax similar to the standard CSS spec https://developer.mozilla.org/en-US/docs/Web/CSS/padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  padding: shorthandValidator,
  /**
   * Sets top padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  paddingTop: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets bottom padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  paddingBottom: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets left padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  paddingLeft: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets right padding. One of 'none', 'small-2', 'small-1', 'small', 'medium', 'large', 'large+1', 'large+2', 'large+3', 'large+4'.
   */
  paddingRight: PropTypes.oneOf(arrayOfSpacerSizes),
  /**
   * Sets the display to be inline-block.
   */
  isInlineBlock: PropTypes.bool,
};

const defaultProps = {
  margin: undefined,
  marginTop: SpacerSizes.NONE,
  marginBottom: SpacerSizes.NONE,
  marginRight: SpacerSizes.NONE,
  marginLeft: SpacerSizes.NONE,
  padding: undefined,
  paddingTop: SpacerSizes.NONE,
  paddingBottom: SpacerSizes.NONE,
  paddingLeft: SpacerSizes.NONE,
  paddingRight: SpacerSizes.NONE,
  isInlineBlock: false,
};

const Spacer = ({
  margin,
  marginTop,
  marginBottom,
  marginLeft,
  marginRight,
  padding,
  paddingTop,
  paddingBottom,
  paddingLeft,
  paddingRight,
  isInlineBlock,
  children,
  ...customProps
}) => {
  const marginShorthand = margin ? mapShorthandToObject('margin', margin) : {};
  const paddingShorthand = padding ? mapShorthandToObject('padding', padding) : {};

  const marginAttributes = {
    marginTop,
    marginRight,
    marginBottom,
    marginLeft,
    ...marginShorthand,
  };

  const paddingAttributes = {
    paddingTop,
    paddingRight,
    paddingBottom,
    paddingLeft,
    ...paddingShorthand,
  };

  const SpacerClassNames = cx([
    `margin-top-${SpacerClassMappings[marginAttributes.marginTop]}`,
    `margin-bottom-${SpacerClassMappings[marginAttributes.marginBottom]}`,
    `margin-left-${SpacerClassMappings[marginAttributes.marginLeft]}`,
    `margin-right-${SpacerClassMappings[marginAttributes.marginRight]}`,
    `padding-top-${SpacerClassMappings[paddingAttributes.paddingTop]}`,
    `padding-bottom-${SpacerClassMappings[paddingAttributes.paddingBottom]}`,
    `padding-left-${SpacerClassMappings[paddingAttributes.paddingLeft]}`,
    `padding-right-${SpacerClassMappings[paddingAttributes.paddingRight]}`,
    { 'is-inline-block': isInlineBlock },
    customProps.className,
  ]);

  return <div {...customProps} className={SpacerClassNames}>{children}</div>;
};

const opts = {
  Sizes: SpacerSizes,
};

Spacer.propTypes = propTypes;
Spacer.defaultProps = defaultProps;
Spacer.Opts = opts;

export default Spacer;
