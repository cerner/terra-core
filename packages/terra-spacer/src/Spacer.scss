@import './variables';
@import './mixins';

:local {
  .margin-top-none {
    margin-top: 0;
  }

  .margin-top-small-minus-2 {
    margin-top: var(--terra-spacer-margin-top-small-minus-2, 0.214rem); // 3px
  }

  .margin-top-small-minus-1 {
    margin-top: var(--terra-spacer-margin-top-small-minus-1, 0.357rem); // 5px
  }

  .margin-top-small {
    margin-top: var(--terra-spacer-margin-top-small, 0.5rem); // 7px
  }

  .margin-top-medium {
    margin-top: var(--terra-spacer-margin-top-medium, 0.714rem); // 10px
  }

  .margin-top-large {
    margin-top: var(--terra-spacer-margin-top-large, 0.857rem); // 12px
  }

  .margin-top-large-plus-1 {
    margin-top: var(--terra-spacer-margin-top-large-plus-1, 1.071rem); // 15px
  }

  .margin-top-large-plus-2 {
    margin-top: var(--terra-spacer-margin-top-large-plus-2, 1.429rem); // 20px
  }

  .margin-top-large-plus-3 {
    margin-top: var(--terra-spacer-margin-top-large-plus-3, 2.143rem); // 30px
  }

  .margin-top-large-plus-4 {
    margin-top: var(--terra-spacer-margin-top-large-plus-4, 3.571rem); // 50px
  }

  .margin-bottom-none {
    margin-bottom: 0;
  }

  .margin-bottom-small-minus-2 {
    margin-bottom: var(--terra-spacer-margin-bottom-small-minus-2, 0.214rem); // 3px
  }

  .margin-bottom-small-minus-1 {
    margin-bottom: var(--terra-spacer-margin-bottom-small-minus-1, 0.357rem); // 5px
  }

  .margin-bottom-small {
    margin-bottom: var(--terra-spacer-margin-bottom-small, 0.5rem); // 7px
  }

  .margin-bottom-medium {
    margin-bottom: var(--terra-spacer-margin-bottom-medium, 0.714rem); // 10px
  }

  .margin-bottom-large {
    margin-bottom: var(--terra-spacer-margin-bottom-large, 0.857rem); // 12px
  }

  .margin-bottom-large-plus-1 {
    margin-bottom: var(--terra-spacer-margin-bottom-large-plus-1, 1.071rem); // 15px
  }

  .margin-bottom-large-plus-2 {
    margin-bottom: var(--terra-spacer-margin-bottom-large-plus-2, 1.429rem); // 20px
  }

  .margin-bottom-large-plus-3 {
    margin-bottom: var(--terra-spacer-margin-bottom-large-plus-3, 2.143rem); // 30px
  }

  .margin-bottom-large-plus-4 {
    margin-bottom: var(--terra-spacer-margin-bottom-large-plus-4, 3.571rem); // 50px
  }

  .margin-left-none {
    margin-left: 0;
  }

  .margin-left-small-minus-2 {
    margin-left: var(--terra-spacer-margin-left-small-minus-2, 0.214rem); // 3px
  }

  .margin-left-small-minus-1 {
    margin-left: var(--terra-spacer-margin-left-small-minus-1, 0.357rem); // 5px
  }

  .margin-left-small {
    margin-left: var(--terra-spacer-margin-left-small, 0.5rem); // 7px
  }

  .margin-left-medium {
    margin-left: var(--terra-spacer-margin-left-medium, 0.714rem); // 10px
  }

  .margin-left-large {
    margin-left: var(--terra-spacer-margin-left-large, 0.857rem); // 12px
  }

  .margin-left-large-plus-1 {
    margin-left: var(--terra-spacer-margin-left-large-plus-1, 1.071rem); // 15px
  }

  .margin-left-large-plus-2 {
    margin-left: var(--terra-spacer-margin-left-large-plus-2, 1.429rem); // 20px
  }

  .margin-left-large-plus-3 {
    margin-left: var(--terra-spacer-margin-left-large-plus-3, 2.143rem); // 30px
  }

  .margin-left-large-plus-4 {
    margin-left: var(--terra-spacer-margin-left-large-plus-4, 3.571rem); // 50px
  }

  .margin-right-none {
    margin-right: 0;
  }

  .margin-right-small-minus-2 {
    margin-right: var(--terra-spacer-margin-right-small-minus-2, 0.214rem); // 3px
  }

  .margin-right-small-minus-1 {
    margin-right: var(--terra-spacer-margin-right-small-minus-1, 0.357rem); // 5px
  }

  .margin-right-small {
    margin-right: var(--terra-spacer-margin-right-small, 0.5rem); // 7px
  }

  .margin-right-medium {
    margin-right: var(--terra-spacer-margin-right-medium, 0.714rem); // 10px
  }

  .margin-right-large {
    margin-right: var(--terra-spacer-margin-right-large, 0.857rem); // 12px
  }

  .margin-right-large-plus-1 {
    margin-right: var(--terra-spacer-margin-right-large-plus-1, 1.071rem); // 15px
  }

  .margin-right-large-plus-2 {
    margin-right: var(--terra-spacer-margin-right-large-plus-2, 1.429rem); // 20px
  }

  .margin-right-large-plus-3 {
    margin-right: var(--terra-spacer-margin-right-large-plus-3, 2.143rem); // 30px
  }

  .margin-right-large-plus-4 {
    margin-right: var(--terra-spacer-margin-right-large-plus-4, 3.571rem); // 50px
  }

  .padding-top-none {
    padding-top: 0;
  }

  .padding-top-small-minus-2 {
    padding-top: var(--terra-spacer-padding-top-small-minus-2, 0.214rem); // 3px
  }

  .padding-top-small-minus-1 {
    padding-top: var(--terra-spacer-padding-top-small-minus-1, 0.357rem); // 5px
  }

  .padding-top-small {
    padding-top: var(--terra-spacer-padding-top-small, 0.5rem); // 7px
  }

  .padding-top-medium {
    padding-top: var(--terra-spacer-padding-top-medium, 0.714rem); // 10px
  }

  .padding-top-large {
    padding-top: var(--terra-spacer-padding-top-large, 0.857rem); // 12px
  }

  .padding-top-large-plus-1 {
    padding-top: var(--terra-spacer-padding-top-large-plus-1, 1.071rem); // 15px
  }

  .padding-top-large-plus-2 {
    padding-top: var(--terra-spacer-padding-top-large-plus-2, 1.429rem); // 20px
  }

  .padding-top-large-plus-3 {
    padding-top: var(--terra-spacer-padding-top-large-plus-3, 2.143rem); // 30px
  }

  .padding-top-large-plus-4 {
    padding-top: var(--terra-spacer-padding-top-large-plus-4, 3.571rem); // 50px
  }

  .padding-bottom-none {
    padding-bottom: 0;
  }

  .padding-bottom-small-minus-2 {
    padding-bottom: var(--terra-spacer-padding-bottom-small-minus-2, 0.214rem); // 3px
  }

  .padding-bottom-small-minus-1 {
    padding-bottom: var(--terra-spacer-padding-bottom-small-minus-1, 0.357rem); // 5px
  }

  .padding-bottom-small {
    padding-bottom: var(--terra-spacer-padding-bottom-small, 0.5rem); // 7px
  }

  .padding-bottom-medium {
    padding-bottom: var(--terra-spacer-padding-bottom-medium, 0.714rem); // 10px
  }

  .padding-bottom-large {
    padding-bottom: var(--terra-spacer-padding-bottom-large, 0.857rem); // 12px
  }

  .padding-bottom-large-plus-1 {
    padding-bottom: var(--terra-spacer-padding-bottom-large-plus-1, 1.071rem); // 15px
  }

  .padding-bottom-large-plus-2 {
    padding-bottom: var(--terra-spacer-padding-bottom-large-plus-2, 1.429rem); // 20px
  }

  .padding-bottom-large-plus-3 {
    padding-bottom: var(--terra-spacer-padding-bottom-large-plus-3, 2.143rem); // 30px
  }

  .padding-bottom-large-plus-4 {
    padding-bottom: var(--terra-spacer-padding-bottom-large-plus-4, 3.571rem); // 50px
  }

  .padding-left-none {
    padding-left: 0;
  }

  .padding-left-small-minus-2 {
    padding-left: var(--terra-spacer-padding-left-small-minus-2, 0.214rem); // 3px
  }

  .padding-left-small-minus-1 {
    padding-left: var(--terra-spacer-padding-left-small-minus-1, 0.357rem); // 5px
  }

  .padding-left-small {
    padding-left: var(--terra-spacer-padding-left-small, 0.5rem); // 7px
  }

  .padding-left-medium {
    padding-left: var(--terra-spacer-padding-left-medium, 0.714rem); // 10px
  }

  .padding-left-large {
    padding-left: var(--terra-spacer-padding-left-large, 0.857rem); // 12px
  }

  .padding-left-large-plus-1 {
    padding-left: var(--terra-spacer-padding-left-large-plus-1, 1.071rem); // 15px
  }

  .padding-left-large-plus-2 {
    padding-left: var(--terra-spacer-padding-left-large-plus-2, 1.429rem); // 20px
  }

  .padding-left-large-plus-3 {
    padding-left: var(--terra-spacer-padding-left-large-plus-3, 2.143rem); // 30px
  }

  .padding-left-large-plus-4 {
    padding-left: var(--terra-spacer-padding-left-large-plus-4, 3.571rem); // 50px
  }

  .padding-right-none {
    padding-right: 0;
  }

  .padding-right-small-minus-2 {
    padding-right: var(--terra-spacer-padding-right-small-minus-2, 0.214rem); // 3px
  }

  .padding-right-small-minus-1 {
    padding-right: var(--terra-spacer-padding-right-small-minus-1, 0.357rem); // 5px
  }

  .padding-right-small {
    padding-right: var(--terra-spacer-padding-right-small, 0.5rem); // 7px
  }

  .padding-right-medium {
    padding-right: var(--terra-spacer-padding-right-medium, 0.714rem); // 10px
  }

  .padding-right-large {
    padding-right: var(--terra-spacer-padding-right-large, 0.857rem); // 12px
  }

  .padding-right-large-plus-1 {
    padding-right: var(--terra-spacer-padding-right-large-plus-1, 1.071rem); // 15px
  }

  .padding-right-large-plus-2 {
    padding-right: var(--terra-spacer-padding-right-large-plus-2, 1.429rem); // 20px
  }

  .padding-right-large-plus-3 {
    padding-right: var(--terra-spacer-padding-right-large-plus-3, 2.143rem); // 30px
  }

  .padding-right-large-plus-4 {
    padding-right: var(--terra-spacer-padding-right-large-plus-4, 3.571rem); // 50px
  }

  .is-inline-block {
    display: inline-block;
  }
}
