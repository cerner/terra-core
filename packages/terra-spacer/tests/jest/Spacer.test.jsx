import React from 'react';
import Spacer from '../../src/Spacer';

describe('Spacer', () => {
  // Snapshot Tests
  it('should render a default component', () => {
    const wrapper = render(<Spacer><div>Test Component</div></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  // Prop Tests
  it('should render a spacer component with small top margin', () => {
    const wrapper = render(<Spacer marginTop="small"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with medium bottom margin', () => {
    const wrapper = render(<Spacer marginBottom="medium"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with large left padding', () => {
    const wrapper = render(<Spacer paddingLeft="large"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with inline-block display', () => {
    const wrapper = render(<Spacer isInlineBlock><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with large vertical and small horizontal margin', () => {
    const wrapper = render(<Spacer margin="large small"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with all margin props set to the given values', () => {
    const wrapper = render((
      <Spacer marginTop="small" marginBottom="small-1" marginLeft="small-2" marginRight="medium">
        <p>Test</p>
      </Spacer>
    ));
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with all padding props set to the given values', () => {
    const wrapper = render((
      <Spacer paddingTop="large+1" paddingBottom="large+2" paddingLeft="large+3" paddingRight="large+4">
        <p>Test</p>
      </Spacer>
    ));
    expect(wrapper).toMatchSnapshot();
  });

  // Custom Prop Test
  it('it should pass in a custom prop', () => {
    const wrapper = render(<Spacer className="spacer"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  // Shorthand Mapping Tests
  it('should render a spacer component with top, right, bottom, and left margin set to large', () => {
    const wrapper = render(<Spacer margin="large"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with vertical margin set to large and horizontal margin set to small', () => {
    const wrapper = render(<Spacer margin="large small"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with vertical top margin set to large, horizontal margin set to medium, and bottom margin set to small', () => {
    const wrapper = render(<Spacer margin="large medium small"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with vertical top margin set to large, right margin set to medium, bottom margin set to small, and left margin set to small-1 ', () => {
    const wrapper = render(<Spacer margin="large medium small small-1"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with top, right, bottom, and left padding set to large', () => {
    const wrapper = render(<Spacer padding="large"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with vertical padding set to large and horizontal padding set to small', () => {
    const wrapper = render(<Spacer padding="large small"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with vertical top padding set to large, horizontal padding set to medium, and bottom padding set to small', () => {
    const wrapper = render(<Spacer padding="large medium small"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a spacer component with vertical top padding set to large, right padding set to medium, bottom padding set to small, and left padding set to small-1 ', () => {
    const wrapper = render(<Spacer padding="large medium small small-1"><p>Test</p></Spacer>);
    expect(wrapper).toMatchSnapshot();
  });
});
