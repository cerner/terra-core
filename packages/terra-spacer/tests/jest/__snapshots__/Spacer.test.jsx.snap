// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer correctly applies the theme context className 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-none padding-right-none orion-fusion-theme"
>
  <div>
    Test Component
  </div>
</div>
`;

exports[`Spacer it should pass in a custom prop 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-none padding-right-none spacer"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a default component 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <div>
    Test Component
  </div>
</div>
`;

exports[`Spacer should render a spacer component with all margin props set to the given values 1`] = `
<div
  class="margin-top-small margin-bottom-small-minus-1 margin-left-small-minus-2 margin-right-medium padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with all padding props set to the given values 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-large-plus-1 padding-bottom-large-plus-2 padding-left-large-plus-3 padding-right-large-plus-4"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with inline-block display 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-none padding-right-none is-inline-block"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with large left padding 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-large padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with large vertical and small horizontal margin 1`] = `
<div
  class="margin-top-large margin-bottom-large margin-left-small margin-right-small padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with medium bottom margin 1`] = `
<div
  class="margin-top-none margin-bottom-medium margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with small top margin 1`] = `
<div
  class="margin-top-small margin-bottom-none margin-left-none margin-right-none padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with top, right, bottom, and left margin set to large 1`] = `
<div
  class="margin-top-large margin-bottom-large margin-left-large margin-right-large padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with top, right, bottom, and left padding set to large 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-large padding-bottom-large padding-left-large padding-right-large"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with vertical margin set to large and horizontal margin set to small 1`] = `
<div
  class="margin-top-large margin-bottom-large margin-left-small margin-right-small padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with vertical padding set to large and horizontal padding set to small 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-large padding-bottom-large padding-left-small padding-right-small"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with vertical top margin set to large, horizontal margin set to medium, and bottom margin set to small 1`] = `
<div
  class="margin-top-large margin-bottom-small margin-left-medium margin-right-medium padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with vertical top margin set to large, right margin set to medium, bottom margin set to small, and left margin set to small-1  1`] = `
<div
  class="margin-top-large margin-bottom-small margin-left-small-minus-1 margin-right-medium padding-top-none padding-bottom-none padding-left-none padding-right-none"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with vertical top padding set to large, horizontal padding set to medium, and bottom padding set to small 1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-large padding-bottom-small padding-left-medium padding-right-medium"
>
  <p>
    Test
  </p>
</div>
`;

exports[`Spacer should render a spacer component with vertical top padding set to large, right padding set to medium, bottom padding set to small, and left padding set to small-1  1`] = `
<div
  class="margin-top-none margin-bottom-none margin-left-none margin-right-none padding-top-large padding-bottom-small padding-left-small-minus-1 padding-right-medium"
>
  <p>
    Test
  </p>
</div>
`;
